bundles/org.openhab.extensionservice.marketplace/src/main/java/org/openhab/extensionservice/marketplace/internal/MarketplaceProxy.java

```diff
@@ -81,24 +82,27 @@ public MarketplaceProxy() {
     public synchronized void refresh() {
         XmlDocumentReader<Marketplace> reader = new MarketplaceXMLReader();
         try {
+            @Nullable
             Marketplace result = reader.readFromXML(url);
-            cachedNodes = result.categories[0].nodes;
+            if (result != null) {
+                cachedNodes.clear();
+                cachedNodes.addAll(Arrays.asList(result.categories[0].nodes));
```

Comment:
There is a slight chance that `getNodes()` is called in between the `clear()` and `addAll(..)`. Perhaps we should synchronize the critical sections on `cachedNodes` so it's thread safe and still performs well.

Reply:
Why couldn't you keep cacheNodes a non-null array but just have it initialized to `new Node[0]`?

Reply:
I've changed it back again. It seems like using an array here has an edge over using a collection. Usually it's the other way around so you can make the variable final to prevent null assignments and use methods like `.isEmpty()` instead of comparing `.length` to 0 etc.